sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/common/BootArgsIndexer.java

```diff
@@ -158,6 +178,14 @@ public static void build(Map<String, Object> argsMap) {
         serviceName = argsMap.get(CommonConstant.SERVICE_NAME_KEY).toString();
 
         instanceId = UUID.randomUUID().toString();
+
+        artifact = argsMap.get(CommonConstant.ARTIFACT_NAME_KEY).toString();
+
+        String name = ManagementFactory.getRuntimeMXBean().getName();
+
+        processId = name.split("@")[0];
```

Comment:
I think 184 and 186 lines are better extracted as functions.

Reply:
done